debezium-api/src/main/java/io/debezium/spi/topic/TopicNamingStrategy.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright Debezium Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.debezium.spi.topic;
+
+import java.util.Map;
+
+import io.debezium.common.annotation.Incubating;
+import io.debezium.spi.schema.DataCollectionId;
+
+/**
+ * An interface that defines the topic naming strategy, including DataChange, SchemaChange, Transaction, Heartbeat events etc.
+ *
+ * @param <I>
+ * @author Harvey Yue
+ */
+@Incubating
+public interface TopicNamingStrategy<I extends DataCollectionId> {
+
+    void configure(Map<String, String> props);
+
+    String dataChangeTopic(I id);
+
+    String schemaChangeTopic();
+
+    String heartbeatTopic();
+
+    String transactionTopic();
+
+    void setLogicalName(String logicName);
+
+    String sanitizedTopicName(String topicName);
```

Comment:
Does the naming strategy have to be responsible for sanitization? As discussed in [DBZ-4180](https://issues.redhat.com/browse/DBZ-4180), how the topic name is sanitized pretty much doesn't depend on how it's built.

The sanitization logic is primarily defined by the topic naming constraints of Kafka, not the user preferences.